static/app/components/events/viewHierarchy/wireframe.tsx

```diff
@@ -1,62 +1,93 @@
-import {useCallback, useEffect, useMemo, useRef, useState} from 'react';
+import {useCallback, useEffect, useLayoutEffect, useMemo, useState} from 'react';
+import {useTheme} from '@emotion/react';
 import styled from '@emotion/styled';
-import {useResizeObserver} from '@react-aria/utils';
+import {mat3, vec2} from 'gl-matrix';
 
 import {ViewHierarchyWindow} from 'sentry/components/events/viewHierarchy';
-import {useMousePan} from 'sentry/components/events/viewHierarchy/useMousePan';
+import {defined} from 'sentry/utils';
+import {Rect, watchForResize} from 'sentry/utils/profiling/gl/utils';
+
+function useResizeCanvasObserver(canvases: (HTMLCanvasElement | null)[]): Rect {
+  const [bounds, setCanvasBounds] = useState<Rect>(Rect.Empty());
+
+  useLayoutEffect(() => {
+    if (!canvases.length) {
+      return undefined;
+    }
+
+    if (canvases.some(c => c === null)) {
+      return undefined;
+    }
+
+    const observer = watchForResize(canvases as HTMLCanvasElement[], entries => {
+      const contentRect =
+        entries[0].contentRect ?? entries[0].target.getBoundingClientRect();
+
+      setCanvasBounds(
+        new Rect(
+          contentRect.x,
+          contentRect.y,
+          contentRect.width * window.devicePixelRatio,
+          contentRect.height * window.devicePixelRatio
+        )
+      );
+    });
+
+    return () => {
+      observer.disconnect();
+    };
+  }, [canvases]);
+
+  return bounds;
+}
 
-const RECT_FILL_ALPHA = 0.005;
-const RECT_OUTLINE_ALPHA = 1;
 const MIN_BORDER_SIZE = 20;
 
-type Rect = Pick<ViewHierarchyWindow, 'x' | 'y' | 'width' | 'height'>;
+interface ViewNode {
+  node: ViewHierarchyWindow;
+  rect: Rect;
+}
 
-export function getCoordinates(hierarchies: ViewHierarchyWindow[]) {
-  return hierarchies.map(hierarchy => {
-    const newHierarchy: Rect[] = [];
+export function getHierarchyDimensions(hierarchies: ViewHierarchyWindow[]): {
+  maxHeight: number;
+  maxWidth: number;
+  nodes: ViewNode[];
+} {
+  const nodes: ViewNode[] = [];
+  const queue: [Rect | null, ViewHierarchyWindow][] = [];
+  [...hierarchies].reverse().forEach(root => queue.push([null, root]));
```

Comment:
Why do you need to reverse this?

Reply:
Right now I wanted to make the assumption that nodes that come later in the hierarchy are drawn overtop of the previous ones and since we pop off of the end and push to our result list, I need to reverse the list so the pop lets us process the first nodes

It's not perfect but the result is more aligned with how the screenshots look when elements over overlapping

Reply:
What if you draw in reverse then? It achieves the same thing but you avoid a shallow copy + array reverse operation
```
let i = arr.length;
while(i--){
  draw(rect[i])
}

// or just for loop if you want
for(let i = arr.length; i >= 0; i--) {
  draw(rect[i])
}
```

Reply:
I went with your suggestion of iterating in reverse order to push onto the list instead of doing a copy. I did this for the initial root order (multiple roots is going to be a rare event anyways but the first root should always be the main application) and also for iterating through the children as you suggested below